plugins/Monolog/tests/Integration/LogTest.php

```diff
@@ -279,9 +286,17 @@ private function recreateLogSingleton($backend, $level = 'INFO')
             'ini.log.logger_file_path' => self::getLogFileLocation(),
             Log\LoggerInterface::class => \Piwik\DI::get(Log\Logger::class),
             'Tests.log.allowAllHandlers' => true,
+            'define_backtrace_constant' => function () use ($forcePrintBacktrace) {
+                $GLOBALS['PIWIK_PRINT_ERROR_BACKTRACE'] = $forcePrintBacktrace;
+
+                return true;
+            },
         ));
         $newEnv->init();
 
+        // set backtrace global via a dummy get
+        $newEnv->getContainer()->get('define_backtrace_constant');
+
```

Comment:
Isn't that a lot more complex as it would need to be? Directly setting the GLOBALS should also work, doesn't it?
```suggestion
        ));
        $newEnv->init();

        $GLOBALS['PIWIK_PRINT_ERROR_BACKTRACE'] = $forcePrintBacktrace;

```

Reply:
Yep, it seems to work without the environment callback. Not sure why it didn't work for me before, perhaps I was setting it too early or I was trying with the constant before. Thanks for double-checking.